tests/unit/alp_cholesky.cpp

```diff
@@ -22,71 +22,169 @@
+	alp::set( TMP, zero_scalar );
+
+	// LLT = -LLT
+	alp::Scalar< T > alpha( -1 );
+	rc = rc ? rc : alp::eWiseLambda(
+		[ ]( const size_t i, const size_t j, T &val ) {
+			(void)i;
+			(void)j;
+			val = -val;
+		},
+		LLT
+	);
+#ifdef DEBUG
+	print_matrix( " << -LLT  >> ", LLT );
+#endif
+
+	// TMP = H - LLT
+	rc = rc ? rc : alp::eWiseApply(
+		TMP, H, LLT,
+		ring.getAdditiveMonoid()
+	);
+#ifdef DEBUG
+	print_matrix( " << H - LLT  >> ", TMP );
+#endif
+
+	//Frobenius norm
+	T fnorm = 0;
+	rc = rc ? rc : alp::eWiseLambda(
+		[ &fnorm ]( const size_t i, const size_t j, T &val ) {
+			(void)i;
+			(void)j;
+			fnorm += val*val;
+		},
+		TMP
+	);
+	fnorm = std::sqrt(fnorm);
+	std::cout << " FrobeniusNorm(H-LL^T) = " << fnorm << "\n";
+	if ( tol < fnorm ) {
+		throw std::runtime_error(
+			"The Frobenius norm is too large. "
+			"Make sure that you have used SPD matrix as input."
+		);
+	}
 
-	alp::utils::MatrixFileReader<
-		double
-	> parser_A( std::string("/home/d/Repos/graphblas/datasets/mymatrix.mtx") );
+	return rc;
+}
 
-	for ( auto it = parser_A.begin() ; it != parser_A.end() ; ++it  ) {
-		std::cout << " i,j,v= " << it.i() << " " << it.j() << " " << it.v() << "\n";
-	}
+void alp_program( const inpdata &unit, alp::RC &rc ) {
+	rc = SUCCESS;
+	auto fname = unit.fname;
+	alp::utils::MatrixFileReader< ScalarType > parser_A( fname );
+	size_t N = parser_A.n();
+
+	alp::Semiring<
+		alp::operators::add< ScalarType >,
+		alp::operators::mul< ScalarType >,
+		alp::identities::zero,
+		alp::identities::one
+	> ring;
+	alp::Scalar< ScalarType > zero_scalar( ring.getZero< ScalarType >() );
 
-	return ;
+	std::cout << "\tTesting ALP cholesky\n"
+		"\tH = L * L^T\n";
 
+	alp::Matrix< ScalarType, structures::Symmetric, Dense > H( N, N );
+	alp::Matrix< ScalarType, structures::UpperTriangular, Dense > L( N, N );
 
+	if( !parser_A.isSymmetric() ) {
+		std::cout << "Symmetric matrix epxected as input!\n";
+		rc = ILLEGAL;
+		return;
+	}
 
-	alp::Semiring< alp::operators::add< double >, alp::operators::mul< double >, alp::identities::zero, alp::identities::one > ring;
+	rc = rc ? rc : alp::buildMatrix( H, parser_A.begin(), parser_A.end() );
 
-	std::cout << "\tTesting ALP cholesky\n"
-	             "\tH = L * L^T\n";
+	if( !internal::getInitialized( H ) ) {
+		std::cout << " Matrix H is not initialized\n";
+	}
 
-	// dimensions of sqare matrices H, L
-	size_t N = 10 * unit;
+#ifdef DEBUG
+	print_matrix( std::string(" << H >> "), H);
+	print_matrix( std::string(" << L >> "), L);
+#endif
 
-	// dimensions of views over H, Q and R
-	size_t n = 2 * unit;
+	rc = rc ? rc : alp::set( L, zero_scalar	);
 
-	alp::Matrix< double, structures::SymmetricPositiveDefinite, Dense > H( N, N );
-	alp::Matrix< double, structures::LowerTriangular, Dense > L( N, N );
+	if( !internal::getInitialized( L ) ) {
+		std::cout << " Matrix L is not initialized\n";
+	}
 
-	rc = algorithms::cholesky_lowtr( L, H, ring );
+	rc = rc ? rc : algorithms::cholesky_lowtr( L, H, ring );
```

Comment:
Same comment as for function definition: Consider renaming as it is operating on an upper-triangular matrix.